test/constructorio-synonyms-one-way.js

```diff
@@ -0,0 +1,587 @@
+    });
+
+    it('should retrieve a list of one way synonyms when supplying a valid phrase', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getOneWaySynonym({
+        phrase: oneWaySynonymPhrase,
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response).to.have.property('one_way_synonym_relations').an('array').length(1);
+        done();
+      });
+    });
+
+    it('should return no results when supplying a non-existent phrase', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getOneWaySynonym({
+        phrase: 1,
+      }, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response).to.have.property('one_way_synonym_relations').an('array').length(0);
+        done();
+      });
+    });
+
+    it('should return no results when not supplying a phrase', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.getOneWaySynonym({}, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response).to.have.property('one_way_synonym_relations').an('array').length(0);
+        done();
+      });
+    });
+
+    it('should return an error when supplying an invalid key/token', (done) => {
+      const constructorio = new Constructorio({
+        apiToken: 'bad-token',
+        apiKey: 'bad-key',
+      });
+
+      constructorio.getOneWaySynonym({
+        phrase: oneWaySynonymPhrase,
+      }, (err, response) => {
+        expect(err).to.be.an('object');
+        expect(err).to.have.property('message').to.match(/You have supplied an invalid/);
+        expect(response).to.be.undefined;
+        done();
+      });
+    });
+  });
+
+  describe('removeOneWaySynonyms', () => {
+    const oneWaySynonymPhrases = ['spices', 'utensils', 'snacks'];
+
+    before((done) => {
+      const addPromiseList = oneWaySynonymPhrases.map(addTestOneWaySynonym);
+
+      // Create test one way synonyms for use in tests
+      Promise.all(addPromiseList).then(() => done()).catch(() => done());
+    });
+
+    after((done) => {
+      const removePromiseList = oneWaySynonymPhrases.map(removeTestOneWaySynonym);
+
+      // Clean up test one way synonyms
+      Promise.all(removePromiseList).then(() => done()).catch(() => done());
+    });
+
+    it('should start removal of all one way synonyms', (done) => {
+      const constructorio = new Constructorio(testConfig);
+
+      constructorio.removeOneWaySynonyms({}, (err, response) => {
+        expect(err).to.be.undefined;
+        expect(response).to.be.an('object');
+        expect(response).to.have.property('message', 'We\'ve started deleting all of your synonym groups. This may take some time to complete.');
```

Comment:
"synonym groups"? language issue on back-end, or does it actually say this?

Reply:
Yea, it actually does say that. And it seems like they don't differentiate between groups and one way synonyms. https://github.com/Constructor-io/autocomplete/blob/6b6441a0a7e4ef76e7cad0116f833fb682586655/common/src/common/api/messages.py#L23